app/services/etherscan_service.rb

```diff
@@ -13,10 +13,30 @@ def is_proof_member?(address)
       address: address
     )
 
-    if response.dig('result')&.last&.dig('to')&.downcase == address.downcase
+    transactions = response.dig('result')
+
+    inCount = 0
+    outCount = 0
+
+    transactions.each do |child|
+      toAddress = child.dig('to').downcase
+      fromAddress = child.dig('from').downcase
+
+
+      if toAddress == address.downcase
+        inCount += 1
+      end
+      if fromAddress == address.downcase
+        outCount += 1
+      end
+    end
```

Comment:
in ruby this can be written like so:
```ruby
      if to_address == address.downcase
        in_count += 1
      elsif from_address == address.downcase
        out_count += 1
      end
    end
```